packages/overlay/src/integrations/sentry/components/traces/TraceList.tsx

```diff
@@ -1,149 +1,92 @@
-import { useEffect, useRef, useState } from "react";
-import { Link, useParams } from "react-router-dom";
-import { Badge } from "~/ui/badge";
+import { useEffect, useRef } from "react";
 import CardList from "../../../../components/CardList";
-import TimeSince from "../../../../components/TimeSince";
-import classNames from "../../../../lib/classNames";
-import { useSpotlightContext } from "../../../../lib/useSpotlightContext";
-import { useSentryTraces } from "../../data/useSentrySpans";
-import useTraceFiltering from "../../hooks/useTraceFiltering";
 import useSentryStore from "../../store";
-import { isLocalTrace } from "../../store/helpers";
-import { getFormattedSpanDuration } from "../../utils/duration";
-import { truncateId } from "../../utils/text";
+import type { Trace } from "../../types";
 import AITranscription from "../insights/aiTraces/AITranscription";
 import { hasAISpans } from "../insights/aiTraces/sdks/aiLibraries";
 import HiddenItemsButton from "../shared/HiddenItemsButton";
-import { TraceRootTxnName } from "./TraceDetails/components/TraceRootTxnName";
 import TraceTreeview from "./TraceDetails/components/TraceTreeview";
-import TraceIcon from "./TraceIcon";
-import TraceListFilter from "./TraceListFilter";
+import TraceItem from "./TraceItem";
 
 type TraceListProps = {
-  aiMode: boolean;
+  onTraceSelect?: (traceId: string) => void;
+  selectedTraceId?: string;
+  traceData: {
+    filtered: Trace[];
+    all: Trace[];
+    visible: Trace[];
+  };
+  displayConfig: {
+    aiMode: boolean;
+    hideSelectedInline?: boolean;
+  };
+  onShowAll: () => void;
 };
 
-export default function TraceList({ aiMode }: TraceListProps) {
-  const { traceId, spanId } = useParams<{ traceId: string; spanId: string }>();
-  const { allTraces, localTraces } = useSentryTraces();
-  const context = useSpotlightContext();
+export default function TraceList({
+  onTraceSelect,
+  selectedTraceId,
+  traceData,
+  displayConfig,
+  onShowAll,
+}: TraceListProps) {
   const getTraceById = useSentryStore(state => state.getTraceById);
   const selectedTraceRef = useRef<HTMLDivElement>(null);
 
-  const [showAll, setShowAll] = useState(!context.experiments["sentry:focus-local-events"]);
-  const visibleTraces = showAll ? allTraces : localTraces;
-  const hiddenItemCount = allTraces.length - visibleTraces.length;
+  const hiddenItemCount = traceData.all.length - traceData.visible.length;
```

Comment:
I'd say this information should also come from `traceData`? Like `traceData.hiddenItemCount`